tests/test_lms_api.py

```diff
@@ -176,11 +193,114 @@ def test_get_remote_id():
             {"username": "DarthVadar", "remote_id": "JamesEarlJones"},
         ]
     }
-    responses.add(  # pylint: disable=no-member
-        responses.GET,  # pylint: disable=no-member
+    responses.add(
+        responses.GET,
         _url("third_party_auth", "providers/{provider}/users/{user}".format(provider=provider_id, user=username)),
         json=expected_response,
     )
     client = lms_api.ThirdPartyAuthApiClient()
     actual_response = client.get_remote_id(provider_id, username)
     assert actual_response == "LukeIamYrFather"
+
+
+def test_jwt_lms_api_client_locally_raises():
+    with raises(NotConnectedToOpenEdX):
+        client = lms_api.JwtLmsApiClient('user-goes-here')
+        client.connect()
+
+
+@mock.patch('enterprise.lms_api.JwtBuilder', mock.Mock())
+def test_jwt_lms_api_client_refresh_token():
+
+    class JwtLmsApiClientTest(lms_api.JwtLmsApiClient):
+        """
+        Test the JwtLmsApiClient class's expired token refreshing logic.
+        """
+        something_called = 0
+
+        @lms_api.JwtLmsApiClient.refresh_token
+        def something(self):
+            """
+            Tests the refresh_token decorator.
+            """
+            self.something_called += 1
+
+    client = JwtLmsApiClientTest('user-goes-here', expires_in=1)
+    assert client.token_expired(), "Token is expired before connect"
+
+    client.something()
+    assert client.something_called == 1, "Wrapped method was called, token created"
+    assert not client.token_expired(), "Token is not yet expired"
+
+    client.something()
+    assert client.something_called == 2, "Wrapped method was called, using existing token."
+    assert not client.token_expired(), "Token is not yet expired"
+
+    time.sleep(2)
+    assert client.token_expired(), "Token is now expired"
+
+    client.something()
+    assert not client.token_expired(), "Token was refreshed"
+    assert client.something_called == 3, "Wrapped method was called"
+
+
+@responses.activate
+@mock.patch('enterprise.lms_api.JwtBuilder', mock.Mock())
+def test_get_course_grades_not_found():
+    username = "DarthVadar"
```

Comment:
@pomegranited It was Vader, wasn't it? Or is it to avoid copyright infringement? :)